source/fundamentals/transactions.txt

```diff
@@ -45,313 +45,240 @@ you want to execute sequentially. We recommend you reuse
+  active session
 
-.. list-table::
-   :widths: 25 25 50
-   :header-rows: 1
-
-   * - Collection
-     - Operation
-     - Description of the Change
+.. important::
+   
+   You must implement the following code patterns when using this API:
 
-   * - ``orders``
-     - insert
-     - Record the purchase information
+   - Pass the session instance to each operation that
+     you want to run in that session
+   - Implement a ``catch`` block in which you identify
+     server transaction errors and implement error-handling logic
 
-   * - ``customers``
-     - update
-     - Append the order id to associate it with the customer
+The following code demonstrates how to perform a transaction by using
+the Core API:
 
-   * - ``inventory``
-     - update
-     - Subtract quantity of ordered items
+.. literalinclude:: /code-snippets/transactions/txn-simple.js
+   :language: javascript
+   :linenos:
+   :emphasize-lines: 2,4,7,11,15,17 
+   :start-after: begin-core
+   :end-before: end-core
 
-A purchase can fail several ways such as if there's insufficient quantity of
-the item in inventory, if the order couldn't be completed, or if your
-payment system is offline.
+.. important:: Use a Session with the Client That Started It
+   
+   The driver throws an error if you provide a session from one ``MongoClient``
+   instance to a different client instance.
 
-If the payment fails, you can use a transaction to make sure that you
-avoid exposing any partial updates that might cause data consistency issues
-for other operations that depend on that data.
+   For example, the following code generates an
+   ``MongoInvalidArgumentError`` error because it creates
+   a ``ClientSession`` instance from the ``client1`` client, but provides
+   this session to the ``client2`` client for a write operation:
 
-Sample Data
-~~~~~~~~~~~
+   .. code-block:: js
+      :emphasize-lines: 2
+      
+      const session = client1.startSession();
+      client2.db('myDB').collection('myColl').insertOne({ name: 'Jane Eyre' }, { session });
 
-The code examples require the following sample data in the ``testdb``
-database to run the multi-document payment transaction:
+To see a fully runnable example that uses this API, see the
+:ref:`node-usage-core-txn` usage example. 
 
-- A document in the ``customers`` collection that describes a customer and
-  their orders.
-- Documents in the ``inventory`` collection that each track quantity and
-  description of an item.
+.. _node-txn-convenient-api:
 
-The document in the ``customers`` collection in this example contains the
-following:
+Convenient Transaction API
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: json
+The Convenient Transaction API provides the following methods to
+implement transactions:
 
-   { _id: 98765, orders: [] }
+- `withSession() <{+api+}/classes/MongoClient.html#withSession>`__: Runs
+  a given operation within a session. The API handles the creation and
```

Comment:
Suggestion:
I think it would be easier to understand what the "given operation" is by using the same terminology as in the next list item:
E.g. "Runs the callback passed to it within a session."